src/utils/concepts_lite.hpp

```diff
@@ -16,6 +16,13 @@
 #include <type_traits>
 #include <utility>
 
+// This is a class template that is required for doing something like static_assert(false)
+// in constexpr if blocks. Actually writing static_assert(false) will always cause a
+// compilation error, even if it is an unchosen constexpr if block. This is fixed in C++23
+// I think.
+template <class...>
+constexpr std::false_type always_false{};
```

Comment:
I don't think this is necessary if you write the `static_assert`s slightly differently. See above.

Reply:
I think the distinction is that `some_constexpr_that_evaluates_to_false` has to somehow depend on template parameters (at least based on my cursory reading of StackOverflow).

Reply:
But it does, doesn't it?

Reply:
[StackOverflow discussion](https://stackoverflow.com/questions/38304847/how-does-a-failed-static-assert-work-in-an-if-constexpr-false-block), based on that I believe 
```c++
if constexpr (std::is_same<T, int>::value) {
  // do something
} else {
  static_assert(false, "blah"); // Technically ill formed
  static_assert(true == false, "blah"); // Technically ill formed
  static_assert(std::is_same<T, int>::value, "blah"); // valid code, since the argument depends on template parameter T
}
```

Reply:
Ah I see. Ok. :+1: